openedx/core/djangoapps/user_api/accounts/api.py

```diff
@@ -416,6 +408,169 @@ def request_password_change(email, orig_host, is_secure):
         raise UserNotFound
 
 
+def check_username_is_valid(username):
+    """Check whether the username input is valid.
+
+    :param username: The proposed username (unicode).
+    :return: True => if the username passes the validity tests.
+             False => if the username fails to pass the validity tests.
+    """
+    try:
+        _validate_username(username)
+    except (AccountUsernameInvalid, UnicodeError):
+        return False
+    return True
+
+
+def check_username_exists(username):
+    """Check whether the username exists in the system.
+
+    :param username: The proposed username (unicode).
+    :return: True => if the username exists in the system.
+             False => if the username doesn't exist in the system.
+    """
+    try:
+        _validate_username_doesnt_exist(username)
+    except (AccountUsernameAlreadyExists):
+        return True
+    return False
```

Comment:
I'm not currently actually using the `check_X_is_valid` or `check_X_exists` functions, but I wrote them earlier because I felt I had needed them, when in fact I didn't later. Maybe it'll come in handy down the road for someone else.